src/memray/_memray/record_reader.cpp

```diff
@@ -430,8 +430,16 @@ RecordReader::nextRecord()
         }
 
         switch (record_type_and_flags.record_type) {
-            case RecordType::UNINITIALIZED: {
-                // Skip it. All remaining bytes should be 0.
+            case RecordType::OTHER: {
+                switch (static_cast<OtherRecordType>(record_type_and_flags.flags)) {
```

Comment:
I know you don't want to, but I would pull this into another function to reduce the clutter, but I am fine either way. I was wanted to take the opportunity to complain

Reply:
I don't want to - at least, not yet. It's a bit cluttered, but this way all of the record types are handled in one function, which I think is a feature. I don't want to handle the "overflow" ones specially just because they didn't fit in 4 bits. They're regular record types, except that we chose to encode them in a weird way.